src/overlays/actors/ovl_Bg_Jya_1flift/z_bg_jya_1flift.c

```diff
@@ -21,31 +39,162 @@ const ActorInit Bg_Jya_1flift_InitVars = {
+                     this->dyna.actor.id, this->dyna.actor.params);
+    }
+}
+
+void BgJya1flift_InitCollision(Actor* thisx, GlobalContext* globalCtx) {
+    BgJya1flift* this = THIS;
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->dyna.actor, &sCylinderInit);
+    this->dyna.actor.colChkInfo.mass = 0xFF;
+}
+
+void BgJya1flift_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgJya1flift* this = THIS;
+    // 1 F lift
+    osSyncPrintf("(１Ｆリフト)(flag %d)(room %d)\n", (u8)D_808930E0, globalCtx->roomCtx.curRoom.num);
+    this->hasInitialized = 0;
+    if (D_808930E0 != 0) {
+        Actor_Kill(thisx);
+        return;
+    }
+    BgJya1flift_InitDynapoly(this, globalCtx, &D_060004A8, 0);
+    Actor_ProcessInitChain(thisx, sInitChain);
+    BgJya1flift_InitCollision(thisx, globalCtx);
+    if (Flags_GetSwitch(globalCtx, (thisx->params & 0x3F)) != 0) {
+        LINK_AGE_IN_YEARS == YEARS_ADULT ? BgJya1flift_ChangeDirection(this) : func_80892E0C(this);
+    } else {
+        func_80892DB0(this);
+    }
+    thisx->room = -1;
+    D_808930E0 = 1;
+    this->hasInitialized = 1;
+}
+
+void BgJya1flift_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgJya1flift* this = THIS;
+
+    if (this->hasInitialized != 0) {
+        D_808930E0 = 0;
+        Collider_DestroyCylinder(globalCtx, &this->collider);
+        DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+    }
+}
+
+void func_80892DB0(BgJya1flift* this) {
+    this->actionFunc = func_80892DCC;
+    this->dyna.actor.posRot.pos.y = D_80893130[0];
+}
+
+void func_80892DCC(BgJya1flift* this, GlobalContext* globalCtx) {
+    if (Flags_GetSwitch(globalCtx, (this->dyna.actor.params & 0x3F)) != 0) {
+        BgJya1flift_ChangeDirection(this);
+    }
+}
+
+void func_80892E0C(BgJya1flift* this) {
+    this->actionFunc = BgJya1flift_DoNothing;
+    this->dyna.actor.posRot.pos.y = D_80893130[0];
+}
+
+void BgJya1flift_DoNothing(BgJya1flift* this, GlobalContext* globalCtx) {
+}
+
+void BgJya1flift_ChangeDirection(BgJya1flift* this) {
+    this->actionFunc = BgJya1flift_Move;
+    this->direction = (this->direction ^ 1);
+    this->dyna.actor.velocity.y = 0.0f;
+}
+
+void BgJya1flift_Move(BgJya1flift* this, GlobalContext* globalCtx) {
+    f32 tempVelocity;
+
+    Math_ApproxF(&this->dyna.actor.velocity.y, 6.0f, 0.4f);
+    this->dyna.actor.velocity.y < 1.0f ? (tempVelocity = 1.0f) : (tempVelocity = this->dyna.actor.velocity.y);
+    if (fabsf(Math_SmoothScaleMaxMinF(&this->dyna.actor.posRot.pos.y, (D_80893130[this->direction]), 0.5f, tempVelocity,
+                                      1.0f)) < 0.001f) {
+        this->dyna.actor.posRot.pos.y = D_80893130[this->direction];
+        BgJya1flift_ResetMoveDelay(this);
+        Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_BLOCK_BOUND);
+        return;
```

Comment:
If possible this should be an else instead of an early return imo.

Reply:
did this really not work? if so post a diff in discord

Reply:
It worked now.  I don't know why but it did.